frontend/pages/settings/personal-access-tokens/new.tsx

```diff
@@ -21,16 +24,48 @@ import Title from 'src/common/Title'
 import Link from 'src/Link'
 import MessageAlert from 'src/MessageAlert'
 import TokenDialog from 'src/settings/authentication/TokenDialog'
-import { TokenActions, TokenActionsKeys, TokenInterface, TokenScope } from 'types/types'
+import { TokenAction, TokenActionKeys, TokenInterface, TokenScope } from 'types/types'
 import { getErrorMessage } from 'utils/fetcher'
 import { plural } from 'utils/stringUtils'
 
+const [TokenReadAction, TokenWriteAction] = Object.values(TokenAction).reduce<Record<string, TokenActionKeys>[]>(
+  ([readActions, writeActions], action) => {
+    let groupedActions = [readActions, writeActions]
+    const [name, kind] = action.split(':')
+
+    if (kind === 'read') {
+      groupedActions = [{ ...readActions, [name]: action }, writeActions]
+    }
+    if (kind === 'write') {
+      groupedActions = [readActions, { ...writeActions, [name]: action }]
+    }
+
+    return groupedActions
+  },
+  [{}, {}],
+)
+
+const isWriteAction = (action: TokenActionKeys) => {
+  return Object.values(TokenWriteAction).includes(action)
+}
+
+const isReadAction = (action: TokenActionKeys) => {
+  return Object.values(TokenReadAction).includes(action)
+}
+
+const getActionName = (action: TokenActionKeys) => {
+  return action.split(':')[0]
```

Comment:
Might be more readable to change this to:

```
const [ actionName ] = action.split(':')
return actionName
```

But happy to leave as-is for a nice one liner.